Content.Client/Chat/ChatManager.cs

```diff
@@ -104,5 +128,84 @@ private void _onChatBoxTextSubmitted(ChatBox chatBox, string text)
                 }
             }
         }
+
+        private void _onFilterButtonToggled(ChatBox chatBox, Button.ButtonToggledEventArgs e)
+        {
+            // TODO make toggled ALL button flip all button states programatically + visually
+            switch (e.Button.Name)
+            {
+                case "Local":
+                    _Localstate = !_Localstate;
+                    if (_Localstate)
+                    {
+                        _filteredChannels = ChatChannel.Local;
+                        RepopulateChat(filteredHistory);
```

Comment:
Fun fact, literally every branch ends with this line! You can move it out and put it after the switch statement instead of having it in 6 places.

Reply:
Whoops, I actually did that but didn't commit it to this branch apparently. Will fix